codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AwsProtocolUtils.java

```diff
@@ -46,6 +52,26 @@ static void generateUnsignedPayloadSigV4Header(GenerationContext context, Operat
                 });
     }
 
+    /**
+     * Writes a serde function for a set of shapes using the passed visitor.
+     * This will walk the input set of shapes and invoke the visitor for any
+     * members of aggregate shapes in the set.
+     *
+     * @see software.amazon.smithy.typescript.codegen.integration.DocumentShapeSerVisitor
+     * @see software.amazon.smithy.typescript.codegen.integration.DocumentShapeDeserVisitor
+     *
+     * @param context The generation context.
+     * @param shapes A list of shapes to generate serde for, including their members.
+     * @param visitor A ShapeVisitor that generates a serde function for shapes.
+     */
+    static void generateDocumentShapeSerde(GenerationContext context, Set<Shape> shapes, ShapeVisitor<Void> visitor) {
```

Comment:
Somewhat related: the terminology here is a little confusing considering we actually have a `document` shape type in Smithy too. Is there a better term we can use... like `generateDocumentBodyShapeSerde` or something?